src/convertToObject.js

```diff
@@ -6,7 +6,55 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const rawLines = sourceString.split(';');
+
+  // filter lines without letters
+  const noEmptyLines = rawLines
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  const aggregatedLines = noEmptyLines.reduce((prev, line) => {
+    if (line.includes(':')) {
+      prev.push(line);
+    } else {
+      prev[prev.length - 1] += line;
+    }
+
+    return prev;
+  }, []);
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. The `reduce` method used here can be replaced with more appropriate methods like `map` or `filter` to process the array of styles.